### DISCUSSION TITLE
MiniDSP XML config load

### DISCUSSION THREAD

**Discussion by nicko88** [mrene/minidsp-rs#53] [Ideas]
I know there are many people, especially the AVSForum BEQ project who would really appreciate a feature to programmatically load MiniDSP XML config files though your control library.

I just thought I would throw up a feature idea for that here.

Thanks!

_*edit*_
I just noticed you have this feature in your roadmap, great!

https://github.com/mrene/minidsp-rs/projects/1#card-52488755

**Comment by mrene** [comment]
I [worked with BEQDesigner](https://github.com/3ll3d00d/beqdesigner/issues/416) to get a direct integration going, it should be in their latest releases (you may have to look at their betas). Since then, the author has also released [EZBEQ](https://github.com/3ll3d00d/ezbeq) which provides a web interface to load these filters dynamically from a web browser. It internally uses this project to overrides the input filters. 

In their case they are merely changing the input PEQs so it's a lot easier to set them directly instead of merging the entire xml file and importing it afterwards (since people have their room correction / sub eqs on the output chans)

Do you have another use case? I can see an intermediate where we can do a partial import of an xml config without necessarily resetting all settings for all input and output channels, but I haven't seen anyone request it yet.

**Comment by nicko88** [comment]
Ah, thanks for the link to the work.

I did not think about the problem of merging PEQ into a user's existing config.  I can see why that is not ideal.

If I go and integrate with minidsp-rs I will use the method that BEQ has used then to add the input PEQ manually.

They have done some excellent work on the BEQ softwares, but I have some fun ideas about using the audio track and a microphone to automatically detect the movie that's playing (like shazam) and then to load in the appropriate PEQ filters for it automatically.

Thanks again for all your hard work over the years reverse engineering this protocol and creating easy-to-use multiplatform packages!
